elytron/ELY-1921-http-external-mechanism.adoc

```diff
@@ -0,0 +1,77 @@
+= HTTP External Mechanism Support in Elytron
+:author:            Ashley Abdel-Sayed
+:email:             aabdelsa@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This proposal is to add support for the External HTTP mechanism with Elytron. This mechanism will perform no verification,
+but will authenticate a user based on credentials established externally. This will be used to accept a principal which
+is passed by the `REMOTE_USER` attribute in the AJP protocol.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/ELY-1921[ELY-1921]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/ELYWEB-99[ELYWEB-99]
+* https://issues.redhat.com/browse/UNDERTOW-1662[UNDERTOW-1662]
+* https://issues.redhat.com/browse/WFCORE-4853[WFCORE-4853]
+* https://issues.redhat.com/browse/WFLY-13125[WFLY-13125]
+* https://issues.redhat.com/browse/EAP7-1323[EAP7-1323]
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:mkopecky@redhat.com[Marek Kopecky]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[X] QE
+
+=== Affected Projects or Components
+* Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* New HTTP `ExternalAuthenticationMechanism`, `ExternalMechanismFactory`, and `WildFlyElytronHttpExternalProvider` classes
+* Ability to pass `REMOTE_USER` attribute from undertow to Elytron (via elytron-web)
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
```

Comment:
Mod-cluster is mentioned in "Community Documentation" section and "Release Note Content" section. But AFAIU mod-cluster is not changed in this RFE, right? If this is true, can you please mention this information in "Non-Requirements" section?

Reply:
Thanks, I've added a note clarifying this.